pkg/proc/core/linux_core.go

```diff
@@ -29,108 +22,217 @@ const NT_X86_XSTATE elf.NType = 0x202 // Note type for notes containing X86 XSAV
 // NT_AUXV is the note type for notes containing a copy of the Auxv array
 const NT_AUXV elf.NType = 0x6
 
+// NT_FPREGSET is the note type for floating point registers.
+const NT_FPREGSET elf.NType = 0x2
+
 const elfErrorBadMagicNumber = "bad magic number"
 
-// readLinuxAMD64Core reads a core file from corePath corresponding to the executable at
-// exePath. For details on the Linux ELF core format, see:
-// http://www.gabriel.urdhr.fr/2015/05/29/core-file/,
-// http://uhlo.blogspot.fr/2012/05/brief-look-into-core-dumps.html,
-// elf_core_dump in http://lxr.free-electrons.com/source/fs/binfmt_elf.c,
-// and, if absolutely desperate, readelf.c from the binutils source.
-func readLinuxAMD64Core(corePath, exePath string) (*Process, error) {
-	coreFile, err := elf.Open(corePath)
-	if err != nil {
-		if _, isfmterr := err.(*elf.FormatError); isfmterr && (strings.Contains(err.Error(), elfErrorBadMagicNumber) || strings.Contains(err.Error(), " at offset 0x0: too short")) {
-			// Go >=1.11 and <1.11 produce different errors when reading a non-elf file.
-			return nil, ErrUnrecognizedFormat
-		}
-		return nil, err
-	}
-	exe, err := os.Open(exePath)
-	if err != nil {
-		return nil, err
-	}
-	exeELF, err := elf.NewFile(exe)
-	if err != nil {
-		return nil, err
-	}
+// Fetch architecture using exeELF.Machine from core file
+// Refer http://man7.org/linux/man-pages/man5/elf.5.html
+const EM_AARCH64 = 183
+const EM_X86_64 = 62
 
-	if coreFile.Type != elf.ET_CORE {
-		return nil, fmt.Errorf("%v is not a core file", coreFile)
-	}
-	if exeELF.Type != elf.ET_EXEC && exeELF.Type != elf.ET_DYN {
-		return nil, fmt.Errorf("%v is not an exe file", exeELF)
+const (
+	_ARM_FP_HEADER_START = 512
+)
+
+// Copied from golang.org/x/sys/unix.Timeval since it's not available on all
+// systems.
+type LinuxCoreTimeval struct {
```

Comment:
Please limit the intermixing of cosmetic changes (such as moving the declaration of structs around) with functional changes. Reviewing this PR is very hard.

Reply:
@aarzilli will remove cosmetic changes to make it simple for review

Reply:
Thank you.

Reply:
> Thank you.

changes are Done now